### ISSUE TITLE
Legendary update progress bar shows incorrect percentage

### ISSUE THREAD

**Issue by RoguedBear** [Heroic-Games-Launcher/HeroicGamesLauncher#4190]
### Describe the bug

I was updating fortnite and noticed that the progress bar was stuck at 90% since half hour.

I went ahead and inspected the legendary logs and found the percentage to be vastly different from the one shown in ui. (this is shown in my screenshot down below where i paused the update and screenshotted the last entry which corresponds with what is displayed in the ui)

I tried to look into why this could be when other metrics are exactly the same as displayed in logs but different for percentage and found this snippet in the following file:
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/17526abcb39738c93711d7257dfba92f9194906d/src/backend/storeManagers/legendary/games.ts#L465-L473

this is in function `onInstallOrUpdateOutput`, and from what it appears looking through rest of the code, we are parsing all other metrics such as download speed, eta, disk write speed, downloaded bytes, etc, directly from the logs, but for percentage for some reason it is being calculated in the code only.

I am not sure what is wrong with the calculation here, but I do think if we are parsing all other metrics from legendary itself, we should also parse the progress from the log rather than calculating ourselves as the legendary binary would have a better context and calculation for the correct progress, reducing the effort done on heroic side.

### Add logs

```text
< logs added as attachment as there are more than 2k lines and 12k lines for heroic & legendary respectively >
```

### Steps to reproduce

1. put a game on update
2. the progress starts from ~80% (see logs)
3. compare with legendary logs
4. you will see a mismatch between progress reported by heroic & legendary, with legendary one being more correct

### Expected behavior

the percentage value should match exactly what is outputted from legendary

### Screenshots

<img width="1266" alt="image_2024-12-21_02-14-52" src="https://github.com/user-attachments/assets/f593eca9-ab8e-487d-b087-3f3860315f8a" />

### Heroic Version

Latest Stable

### System Information

```
System Information:
CPU: 12x 13th Gen Intel(R) Core(TM) i5-13420H
Memory: 16.89 GB (used: 8.61 GB)
GPUs:
  GPU 0:
    Name: Intel Corporation Raptor Lake-P [UHD Graphics]
    IDs: D=a7a8 V=8086 SD=3c90 SV=17aa
    Driver: 31.0.101.4255
OS: Microsoft Windows 11 Home Single Language 10.0.22631 (win32)

The current system is not a Steam Deck
We are not running inside a Flatpak container

Software Versions:
  Heroic: 2.15.2 Emeth
  Legendary: 0.20.34 Direct Intervention
  gogdl: 1.1.2
  comet: comet 0.1.2

  Nile: 1.1.2 Will A. Zeppeli
```

### Additional information

log files which corroborate the same:
[2024-12-20T19_41_48-heroic.log](https://github.com/user-attachments/files/18215387/2024-12-20T19_41_48-heroic.log)

[2024-12-20T19_41_48-legendary.log](https://github.com/user-attachments/files/18215388/2024-12-20T19_41_48-legendary.log)

**Comment by RoguedBear** [comment]
If the bug is confirmed and if it is alright, i would be willing to open a PR to fix this

**Comment by RoguedBear** [comment]
I was checking this issue again and traced the percentage calculation was added in PR: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/1256 specifically in this commit: 223b91510fdd473c87cc3996eef8a755162b9ba4 by @arielj 

looking at the intent now behind the calculation which is to take in to account already downloaded stuff in case the update is paused/resumed again, I'm confused if this issue would be considered valid.

on one hand, the calculation should take into account only the remaining update download size instead of total game size already downloaded, since the percentage is for the update only

on the other, this feels misleading as i was assuming the update is 90% done, which is not correct as only a few gbs were downloaded of the 15GB update, but in the context of overall progress it makes sense

---

example of perspective:

I am taking rocket league as an example here which i have not updated since ages, and as per legendary logs the update size is 5.5GB
![image](https://github.com/user-attachments/assets/7c3e4852-bc59-4b7e-aa88-a93d22e17bfd)


when I press update, legendary's percentage bar is 0% meanwhile heroic shows 81%

![image](https://github.com/user-attachments/assets/83b2e59a-4ce3-4242-9550-bb05f3d6cf15)

so it is confusing as seeing 81%, one would think the update is 81% downloaded when that is not the case

basically the difference in perspective of saying if 81% done is:
- 81% of 5GB update that is is downloaded
- or 81% of 28GB game that is downloaded

**Comment by Etaash-mathamsetty** [comment]
a contribution to fix this would be welcome :)

**Comment by RoguedBear** [comment]
> a contribution to fix this would be welcome :)

great! i have one question though before I get started:
- should I only make the change to read legendary's progress bar
- or apart from above point should I also make it possible to report the remaining download size to the UI?

**Comment by Etaash-mathamsetty** [comment]
> > a contribution to fix this would be welcome :)
> 
> great! i have one question though before I get started:
> 
>     * should I only make the change to read legendary's progress bar
> 
>     * or apart from above point should I also make it possible to report the remaining download size to the UI?

you probably don't have to change much. I'll just do some quick math here:

current code:
`percentage = (downloaded/total) * 100`
instead just do:
`percentage = (downloaded-initial_downloaded)/(total-inital_downloaded) * 100`

**Comment by RoguedBear** [comment]
oh damn. this is much cleaner! Thanks for the tip. I'll work on this on the weekends if that's alright. 

would it be possible for you to assign this issue to me?

**Comment by arielj** [comment]
Closing, should be handled in legendary https://github.com/Heroic-Games-Launcher/legendary/issues/35
